src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1156,4 +1158,131 @@ function ($a) {
             ->end()
         ;
     }
+
+    private function addHttpClientSection(ArrayNodeDefinition $rootNode)
+    {
+        $subNode = $rootNode
+            ->children()
+                ->arrayNode('http_client')
+                    ->info('Http Client configuration')
+                    ->canBeEnabled()
+                    ->fixXmlConfig('http_client')
+                    ->fixXmlConfig('api_client')
+                    ->children()
+                        ->integerNode('max_host_connections')
+                            ->info('The maximum number of connections to a single host.')
+                            ->defaultValue(6)
+                        ->end()
+                        ->scalarNode('no_proxy')
+                            ->info('A comma separated list of hosts that do not require a proxy to be reached.')
+                            ->defaultNull()
+                        ->end();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                        ->arrayNode('http_clients')
+                            ->useAttributeAsKey('name')
+                            ->normalizeKeys(false)
+                            ->arrayPrototype()
+                                ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                                ->end()
+                            ->end()
+                        ->end()
+                        ->arrayNode('api_clients')
+                            ->useAttributeAsKey('name')
+                            ->normalizeKeys(false)
+                            ->arrayPrototype()
+                                ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    private function addHttpClientOptionsSection(NodeBuilder $rootNode): NodeBuilder
+    {
+        return $rootNode
+            ->arrayNode('default_options')
+                ->children()
+                    ->scalarNode('auth')
+                        ->info('An HTTP Basic authentication "username:password".')
+                    ->end()
+                    ->arrayNode('query')
+                        ->info('Associative array of query string values merged with URL parameters.')
+                        ->normalizeKeys(false)
+                        ->scalarPrototype()->end()
+                    ->end()
+                    ->arrayNode('headers')
+                        ->info('Associative array: header => value(s).')
+                        ->normalizeKeys(false)
+                        ->variablePrototype()->end()
+                    ->end()
+                    ->integerNode('max_redirects')
```

Comment:
Should we put the default values here for the options that define them? For example, `20` for the `max_reidrects`, `true` or `false` for `buffer`, etc.

Reply:
The component already defines defaults for those options. Do we need different ones for the bundle?

Reply:
I agree with @derrabus